### PR TITLE
taxonomy: taking into account the ingredient origin when retrieving the corresponding Ecobalyse code

### PR DESCRIPTION
**Pull request by very-smartin**
### What

Goal : being able to take into account the origin when retrieving the corresponding ecobalyse code for a given ingredient

### CODE REVIEW

File: lib/ProductOpener/Ingredients.pm [hunk 1]
```diff
@@ -3512,6 +3543,35 @@ sub get_missing_ecobalyse_ids ($ingredients_ref) {
...
+
+reference to the name of the country 
+
+=head3 Return values
+
+=head4 $ecobalyse_area
+
+=cut
+
+sub get_geographical_area ($country_code) {
```
Comment (stephanegigandet) [inline lib/ProductOpener/Ingredients.pm:3562]: We use country code to refer to the 2 letters country codes, could you rename $country_code to $originid ? in the origins field you will get things like "en:france" or "en:britanny".

File: lib/ProductOpener/Ingredients.pm [hunk 2]
```diff
@@ -3472,11 +3472,42 @@ sub get_missing_ecobalyse_ids ($ingredients_ref) {
 
 		# List of suffixes
 		my @suffixes = ();
+		# If the ingredient is both organic and French...
+		if ((defined $ingredient_ref->{labels}) and 
+		    ($ingredient_ref->{labels} =~ /\ben:organic\b/) and 
+		    (defined $ingredient_ref->{origins}) and 
+		    (get_geographical_area($ingredient_ref->{origins}) eq "fr")) {
+
+		    push @suffixes, "_labels_en_organic_origins_en_france";
```
Comment (stephanegigandet) [inline lib/ProductOpener/Ingredients.pm:3481]: If get_geographical_area() returns "fr", we need to push both suffixes: the "origins_en_france" one + the "origins_en_european_union_one" as we may not have a FR property in Ecobalyse, but we could have a EU one.

**Comment by very-smartin** [comment]
@stephanegigandet i've got some issues on this code and i don't know how to solve them... What should i do ?

**Review by stephanegigandet — APPROVED** [review]
Perfect, thank you!

**Pull request merged**